src/components/PostDetails.tsx

```diff
@@ -2,105 +1,121 @@
-      <div className="content" data-cy="PostDetails">
-        <div className="block">
-          <h2 data-cy="PostTitle">
-            #18: voluptate et itaque vero tempora molestiae
-          </h2>
+type Props = {
+  activePost: Post;
+};
 
-          <p data-cy="PostBody">
-            eveniet quo quis laborum totam consequatur non dolor ut et est
-            repudiandae est voluptatem vel debitis et magnam
-          </p>
-        </div>
+export const PostDetails: React.FC<Props> = ({ activePost }) => {
+  const [isLoadingComments, setIsLoadingComments] = useState(false);
+  const [isLoadingBtn, setIsLoadingBtn] = useState(false);
+  const [comments, setComments] = useState<Comment[]>([]);
+  const [isError, setIsError] = useState(false);
+  const [isCommentFormVisible, setIsCommentFormVisible] = useState(false);
 
-        <div className="block">
-          <Loader />
+  useEffect(() => {
+    setIsLoadingComments(true);
+    setIsCommentFormVisible(false);
 
-          <div className="notification is-danger" data-cy="CommentsError">
-            Something went wrong
-          </div>
+    getComments(activePost.id)
+      .then(setComments)
+      .catch(() => setIsError(true))
+      .finally(() => setIsLoadingComments(false));
+  }, [activePost]);
 
-          <p className="title is-4" data-cy="NoCommentsMessage">
-            No comments yet
-          </p>
+  const handleDeleteComment = (commentId: number) => {
+    setComments(commentsList =>
+      commentsList.filter(comment => comment.id !== commentId),
+    );
 
-          <p className="title is-4">Comments:</p>
+    deleteComment(commentId).then(() => {});
+  };
 
-          <article className="message is-small" data-cy="Comment">
-            <div className="message-header">
-              <a href="mailto:misha@mate.academy" data-cy="CommentAuthor">
-                Misha Hrynko
-              </a>
-              <button
-                data-cy="CommentDelete"
-                type="button"
-                className="delete is-small"
-                aria-label="delete"
-              >
-                delete button
-              </button>
-            </div>
+  const onAddComment = ({ name, email, body }: CommentData) => {
+    const postId = activePost?.id;
 
-            <div className="message-body" data-cy="CommentBody">
-              Some comment
-            </div>
-          </article>
+    setIsLoadingBtn(true);
+    addComment({ postId, name, email, body })
+      .then(newComment => setComments([...comments, newComment]))
+      .finally(() => setIsLoadingBtn(false));
+  };
 
-          <article className="message is-small" data-cy="Comment">
-            <div className="message-header">
-              <a href="mailto:misha@mate.academy" data-cy="CommentAuthor">
-                Misha Hrynko
-              </a>
+  return (
+    <div className="content" data-cy="PostDetails">
+      <div className="content" data-cy="PostDetails">
```

Comment:
The `data-cy="PostDetails"` attribute is duplicated on this div. Consider removing one of them to avoid confusion.